oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/DependencyTask.groovy

```diff
@@ -16,288 +16,86 @@
 
 package com.google.android.gms.oss.licenses.plugin
 
+import com.android.tools.build.libraries.metadata.AppDependencies
 import groovy.json.JsonBuilder
-import groovy.json.JsonException
-import groovy.json.JsonSlurper
 import org.gradle.api.DefaultTask
-import org.gradle.api.Project
-import org.gradle.api.artifacts.Configuration
-import org.gradle.api.artifacts.ConfigurationContainer
-import org.gradle.api.artifacts.Dependency
-import org.gradle.api.artifacts.ExternalModuleDependency
-import org.gradle.api.artifacts.ProjectDependency
-import org.gradle.api.artifacts.ResolveException
-import org.gradle.api.artifacts.ResolvedArtifact
-import org.gradle.api.artifacts.ResolvedDependency
-import org.gradle.api.tasks.Input
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.InputFile
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
-import org.gradle.internal.component.AmbiguousVariantSelectionException
 import org.slf4j.LoggerFactory
 
-/**
- * This task does the following:
- * First it finds all dependencies that meet all the requirements below:
- * 1. Can be resolved.
- * 2. Not test compile.
- * Then it finds all the artifacts associated with the dependencies.
- * Finally it generates a json file that contains the information about these
- * artifacts.
- */
-class DependencyTask extends DefaultTask {
-    protected Set<String> artifactSet = []
-    protected Set<ArtifactInfo> artifactInfos = []
-    protected static final String LOCAL_LIBRARY_VERSION = "unspecified"
-    private static final String TEST_PREFIX = "test"
-    private static final String ANDROID_TEST_PREFIX = "androidTest"
-    private static final Set<String> TEST_COMPILE = ["testCompile",
-                                                     "androidTestCompile"]
+import java.util.stream.Collectors
 
-    private static final Set<String> PACKAGED_DEPENDENCIES_PREFIXES = ["compile",
-                                                                       "implementation",
-                                                                       "api"]
+import static com.android.tools.build.libraries.metadata.Library.LibraryOneofCase.MAVEN_LIBRARY
 
+/**
+ * Converts the AppDependencies protobuf file provided by the Android Gradle
+ * Plugin into a JSON format that will be consumed by the {@link LicensesTask}.
+ *
+ * If the protobuf is not present (e.g. debug variants) it writes a single
+ * dependency on the {@link DependencyUtil#ABSENT_ARTIFACT}.
+ */
+abstract class DependencyTask extends DefaultTask {
     private static final logger = LoggerFactory.getLogger(DependencyTask.class)
 
-    private Project project
-
     @OutputDirectory
     File outputDir
```

Comment:
although not strictly necessary, I would not use File for these two output properties but instead 
abstract DirectoryProperty outputDir
abstract RegularFileProperty outputFile

now if I understood correctly, outputFile is located inside outputDir and it is the only file present in that folder. If that's the case, then you do not need outputDir at all. 

tagging the 'outputFile' as the output is enough.

Reply:
Done